e2e/util_test.go

```diff
@@ -671,3 +681,36 @@ func (l *fixedTimeSource) Advance(by time.Duration) {
 
 	l.time = l.time.Add(by)
 }
+
+func keysToS3URL(scheme, bucket string, keys ...string) []string {
+	l := len(keys)
+	res := make([]string, 0, l)
+
+	for _, k := range keys {
+		res = append(res, scheme+bucket+"/"+k)
+	}
+
+	return res
+}
+
+func expectedSliceToEqualsMap(expected []string, shouldSort bool) map[int]compareFunc {
+	m := make(map[int]compareFunc)
+
+	if shouldSort {
+		sort.Strings(expected)
+	}
+
+	for i, e := range expected {
+		m[i] = equals(e)
+	}
+	return m
+}
```

Comment:
Instead of this function, you may define `expected` in `TestCompletionFlag` as `map[int]compareFunc` just like the other tests. I think it would make more easier to read the tests as the assertion logic is not that complex to hide it.